torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Dataset splitting utilities."""
+
+from copy import deepcopy
+from itertools import accumulate
+from math import floor
+from typing import List, Optional, Sequence, Tuple, Union
+
+from rtree.index import Index, Property
+from torch import Generator, default_generator, randint, randperm
+
+from ..datasets import GeoDataset
+from .utils import BoundingBox
+
+__all__ = (
+    "random_bbox_assignment",
+    "random_bbox_splitting",
+    "random_grid_cell_assignment",
+    "roi_split",
+    "time_series_split",
+)
+
+
+def _fractions_to_lengths(fractions: Sequence[float], total: int) -> Sequence[int]:
+    """Utility to divide a number into a list of integers according to fractions.
+
```

Comment:
May be useful to add something like:
```rst
Implementation based on :meth:`torch.utils.data.random_split`.
```
both for legal reasons but also for updating someday if we ever find a bug.